crates/router/src/core/payment_methods/cards.rs

```diff
@@ -3031,6 +3031,15 @@ pub async fn list_payment_methods(
                             .connector
                             .connector_name
                             .to_string()
+                        && first_routable_connector
+                            .connector
+                            .merchant_connector_id
+                            .as_ref()
+                            .map(|merchant_connector_id| {
+                                *merchant_connector_id.get_string_repr()
+                                    == intermediate.merchant_connector_id
+                            })
+                            .unwrap_or_default()
```

Comment:
When we are filtering based on the `merchant_connector_id` which is a unique parameter for an mca, why should we also check for the `connector_name` ?